lib/Twig/Extension/Core.php

```diff
@@ -1241,7 +1241,7 @@ function twig_test_empty($value)
         return 0 == count($value);
     }
 
-    if (method_exists($value, '__toString')) {
+    if (is_object($value) && method_exists($value, '__toString')) {
```

Comment:
can be optimized, as the `is_object` test is now always done there is not point for the test at line 1280 (`return '' === $value || false === $value || null === $value || array() === $value;`) if `is_object` returned `true`